### ISSUE TITLE
Problem linking the latest source 

### ISSUE THREAD

**Issue by sorlok** [EasyRPG/Player#2317]
This is a compilation issue, not a game-related issue.

Some time over the past few weeks, the linker stage has started to fail:

```
libEasyRPG_Player.a(audio_midi.cpp.o): In function `std::_MakeUniq<AudioResampler>::__single_object std::make_unique<AudioResampler, std::unique_ptr<AudioDecoder, std::default_delete<AudioDecoder> > >(std::unique_ptr<AudioDecoder, std::default_delete<AudioDecoder> >&&)':
audio_midi.cpp:(.text._ZSt11make_uniqueI14AudioResamplerJSt10unique_ptrI12AudioDecoderSt14default_deleteIS2_EEEENSt9_MakeUniqIT_E15__single_objectEDpOT0_[_ZSt11make_uniqueI14AudioResamplerJSt10unique_ptrI12AudioDecoderSt14default_deleteIS2_EEEENSt9_MakeUniqIT_E15__single_objectEDpOT0_]+0x5d): undefined reference to `AudioResampler::AudioResampler(std::unique_ptr<AudioDecoder, std::default_delete<AudioDecoder> >, AudioResampler::Quality)'
collect2: error: ld returned 1 exit status
CMakeFiles/EasyRPG_Player_exe.dir/build.make:113: recipe for target 'easyrpg-player' failed
```

This is with the following version of gcc:

```
$ gcc --version
gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

Commenting out line 77 in the src/audio_midi.cpp causes the build to pass:

```
        if (mididec && resample) {
//                mididec = std::make_unique<AudioResampler>(std::move(mididec));
        }
```

I have no idea why this is failing; the constructor is clearly there, and it's not a template function (despite taking a templated parameter). My template metaprogramming is a little rusty, but I honestly can't figure out why this is "wrong". 

I am compiling with AUDIO_BACKEND=OFF in case that matters.

Thanks.

**Comment by Ghabry** [comment]
This is fixed in #2305

**Comment by sorlok** [comment]
Great, thanks! I'll close this early since it's a known issue.
